core/chains/evm/txmgr/common.go

```diff
@@ -75,28 +74,16 @@ func batchSendTransactions[ADDR types.Hashable[ADDR], TX_HASH types.Hashable[TX_
 	return reqs, nil
 }
 
-func getGethAddressFromADDR[ADDR commontypes.Hashable[ADDR]](addr ADDR) (common.Address, error) {
-	addrHex, err := addr.MarshalText()
-	if err != nil {
-		return common.Address{}, errors.Wrapf(err, "failed to serialize address to text: %s", addr.String())
+func stringToGethAddress(s string) (common.Address, error) {
+	if !common.IsHexAddress(s) {
+		return common.Address{}, fmt.Errorf("invalid hex address: %s", s)
 	}
-	var gethAddr common.Address
-	err = gethAddr.UnmarshalText(addrHex)
-	if err != nil {
-		return common.Address{}, errors.Wrapf(err, "failed to deserialize address from text: %s. Original address: %s", addrHex, addr.String())
-	}
-	return gethAddr, nil
+	return common.HexToAddress(s), nil
 }
 
-func getGethHashFromHash[HASH commontypes.Hashable[HASH]](hash HASH) (common.Hash, error) {
-	hashHex, err := hash.MarshalText()
-	if err != nil {
-		return common.Hash{}, errors.Wrapf(err, "failed to serialize hash to text: %s", hash.String())
-	}
-	var gethHash common.Hash
-	err = gethHash.UnmarshalText(hashHex)
-	if err != nil {
-		return common.Hash{}, errors.Wrapf(err, "failed to deserialize hash from text: %s. Original hash: %s", hashHex, hash.String())
+func stringToGethHash(s string) (common.Hash, error) {
+	if _, err := hexutil.Decode(s); err != nil {
+		return common.Hash{}, err
```

Comment:
What about this? I think it checks for more, and it's more efficient:
```suggestion
func stringToGethHash(s string) (h common.Hash, err error) {
	err = h.UnmarshalText(s)
	return
```
The same approach might give a more informative error for `common.Address` too.